runtime/shared_common/shrinit.cpp

```diff
@@ -3952,6 +3953,30 @@ j9shr_getCacheSizeBytes(J9JavaVM *vm)
 	return ret;
 }
 
+/**
+ * Determine the type of shared class cache that is enabled. Either the current default, 
+ * Bootstrap Classes Only, or user defined shared cache, enabled via "-Xshareclasses" on the command line. 
+ * 
+ * @param [in] vm Pointer to the VM structure for the JVM
+ *
+ * @return J9SharedClassCacheMode enum that idicates the Shared Class Cache that is in effect
+ * 
+ */
+J9SharedClassCacheMode
+j9shr_getSharedClassCacheMode(J9JavaVM *vm)
+{
+	J9SharedClassCacheMode ret = J9SharedClassCacheUserDefined;
+	/* Only bootstrap classes are shared by default */
+	if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES)) {
+		if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)) {
+			ret = J9SharedClassCacheUserDefined;
+		} else {
+			ret = J9SharedClassCacheBootstrapOnly;
+		}
+	}
```

Comment:
I believe, as your comment says, only bootstrap classes are shared by default, so shouldn't `ret` be initialized to `J9SharedClassCacheBootstrapOnly`? There are three possible values, yet this can't return `J9SharedClassCacheBoostrapAndExtension`; that doesn't seem right.

@hangshao0 Please comment.

Reply:
> yet this can't return J9SharedClassCacheBoostrapAndExtension.

`J9SharedClassCacheBoostrapAndExtension` is a possible case in the future that after we change the default class sharing from "boostrap classes" to "bootstrap + extension classes". User cannot configure the JVM to share "bootstrap + extension classes" now.

Reply:
I agree that it is better to initialize `ret` to `J9SharedClassCacheBootstrapOnly`.

Reply:
I have made that change, is this okay to resolve?